docs/source/validation_rules.rst

```diff
@@ -0,0 +1,714 @@
+-------------------
+
+url
+~~~
+
+- Using the ``url`` rule implies the user should add a URL to a free text box as a string. This function will check that the user has provided a usable URL. It will check for any standard URL error and throw an error if one is found. Further additions to this rule can allow for checking that a specific type of URL is added. For example, if the user needs to add a ``http://protocols.io`` URL, ``http://protocols.io`` can be added after url to perform this check. If the provided url does not contain this specific string, an error will be raised.
+
+- Format:
+
+  - ``url <optional strings> <raised message level>``
+
+    - ``url`` must be specified first then an arbitrary number of strings can be added after (separated by spaces) to add additional levels of specificity.
+
+  - Alternatively, its valid to pass only ``url`` to simply check if the input is a url.
+
+- Examples:
+
+  - ``url http://protocols.io`` Will check that any input is a valid URL, and will also check to see that the URL contains the string ``http://protocols.io`` If not, an error will be raised.
+
+  - ``url dx.doi http://protocols.io`` Will check that any input is a valid URL, and will also check to see that the URL contains the strings ``dx.doi`` and ``http://protocols.io``. If not, an error will be raised.
+
+- Default behavior: raises ``error``
+
+Required Validation Type
+------------------------
+
+required
+~~~~~~~~
+
+
+An attribute's requirement is typically set using the required column (csv) or field (JSONLD) in the data model. A ``True`` value means a users must supply a value, ``False`` means they are allowed to skip providing a value.
+
+Some users may want to use the same attribute across several manifests, but have different requirements based on the manifest/component. For example, say the data model contains an attribute called PatientID, and this attribute is used in manifests Biospecimen, Patient and Demographics. Say the modeler wants to require that PatientID be required in the Patient manifest but not Biospecimen or Demographics. In the standard Data Model format, there is only one requirement option per Attribute, so one would not be able to set requirements per component. But with the advent of component based rule settings, this can now be achieved.
+
+Requirements can be specified per component by setting the required field in the data model to ``False``, and using component based rule setting along with the required "rule".
+
+Note: this new required validation rule is not a traditional validation rule, but rather impacts the JSON validation schema. This means requirements propagate automatically to manifests as well.
+
+Notes:
+
+- When using ``required`` in validation rules, the ``Required`` **column/field must be set to** ``False`` or this will cause the rule to not work as expected (i.e. components were the attribute is expected to not be required due to the validation rules, will still be required).
```

Comment:
You meant that **column/field must be set to** ``False`` in the data model? can we clarify that?